### PR TITLE
Add authorization support for reports endpoint

### PR DESCRIPTION
**Pull request by lzap**
SSIA thanks to @ekohl who noticed this important bug

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -13,7 +13,7 @@ GIT
 PATH
   remote: .
   specs:
-    smart_proxy_host_reports (0.0.3)
+    smart_proxy_host_reports (0.0.4)
       json (>= 2.3.0)
       rack (>= 2.1.0)
```
Comment (lzap) [inline Gemfile.lock]: Does anyone know why on earth is this in the lock file? Do we need this?

Comment (ekohl) [reply Gemfile.lock]: AFAIK we never have lockfiles in plugin repos so I'd just consider to remove the file and gitignore it.

File: settings.d/reports.yml.example [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-:enabled: true
+:enabled: https
```
Comment (ekohl) [inline settings.d/reports.yml.example]: I'd be fine with `true` as a default and let the installer ensure `https`.

File: Gemfile.lock (hunk 1 above)
Comment (lzap) [reply Gemfile.lock]: Done.

File: settings.d/reports.yml.example (hunk 2 above)
Comment (lzap) [reply settings.d/reports.yml.example]: OK

**Comment by lzap** [comment]
Ok rebased on the plugin name change, let me merge this, do a release and I will test this on the production instance. I will make sure this works as expected and fix it in case it does not.

**Review by ofedoren — APPROVED** [review]
Thanks, @lzap. I don't see any other concerns, so mergnig.

**Pull request merged**